### PR TITLE
Migrate remaining amounts in db to ints

### PR DESCRIPTION
**Pull request by asood123**

### CODE REVIEW

**Comment by asood123** [comment]
Don't merge this yet, just approve if you are good with it. Need to sync with website (and maybe even app) and don't want to block pushes

File: server/controllers/donations.js [hunk 1]
```diff
@@ -61,11 +43,11 @@ const stripeDonation = (req, res, next) => {
     return next(new errors.BadRequest('Stripe Token missing.'));
   }
 
-  if (!amountFloat) {
+  if (!amount) {
     return next(new errors.BadRequest('Payment Amount missing.'));
   }
 
-  if (amountInt < 50) {
+  if (amount < 50) {
```
Comment (xdamman) [inline server/controllers/donations.js]: How do you feel about being super explicit and say "amountInCents"? #idea

File: migrations/20170127223729-change-amounts-to-ints-migration.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+    return queryInterface.sequelize.query('update "Transactions" SET amount = amount*100')
+      .then(() => queryInterface.sequelize.query('update "Subscriptions" SET amount = amount*100'))
```
Comment (xdamman) [inline migrations/20170127223729-change-amounts-to-ints-migration.js]: Love it, thanks for tackling this!

File: server/controllers/donations.js [hunk 3]
```diff
@@ -123,11 +105,12 @@ export {stripeDonation as stripe};
 export {stripeDonation as post};
 
 
+/*
 const paypalDonation = (req, res, next) => {
   const { group } = req;
   const { payment } = req.required;
   const currency = payment.currency || group.currency;
-  const amountFloat = payment.amount; // TODO: clean this up when we switch all amounts to INTEGER
+  const amount = parseInt(payment.amount * 100, 10);
```
Comment (xdamman) [inline server/controllers/donations.js]: that's where having a very explicit `amountInCents` would make this more readable.

Should the frontend keep using float?

File: server/controllers/stripeWebhook.js [hunk 4]
```diff
@@ -219,8 +219,7 @@ export default function stripeWebhook(req, res, next) {
         platformFeeInTxnCurrency: fees.applicationFee,
         paymentProcessorFeeInTxnCurrency: fees.stripeFee,
         data: {charge, balanceTransaction},
-        description: 'Recurring subscription', // remove #postmigration
-        tags: ['Donation'], // remove #postmigration
+        description: 'Recurring subscription',
```
Comment (xdamman) [inline server/controllers/stripeWebhook.js]: Can you prepend `monthly` or `yearly`?

File: server/controllers/donations.js (hunk 1 above)
Comment (asood123) [inline server/controllers/donations.js]: By default, all amounts are in cents now. How about the opposite? In cases where it's not (like Paypal expects in dollars), we name it amountNotInCents or amountFloat?

File: server/controllers/donations.js (hunk 3 above)
Comment (asood123) [inline server/controllers/donations.js]: yes, I think the frontend should send this back in cents now as well. I'll flip it. Only time we ever turn it into float is to display or when it leaves our servers.

File: server/controllers/donations.js (hunk 1 above)
Comment (xdamman) [inline server/controllers/donations.js]: As the comment on our API suggests, people by default will assume that "amount" is in float, not in cents. That's why I'd rather be explicit. The more we can avoid "RTFM", the better.

File: server/controllers/donations.js (hunk 3 above)
Comment (xdamman) [inline server/controllers/donations.js]: ok

File: server/controllers/donations.js (hunk 1 above)
Comment (asood123) [inline server/controllers/donations.js]: I get it. It'd be better to do, it's a fair amount of work to rename amount everywhere (lots of tables have it, not to mention about 100s of mentions of it in the code - both backend and frontend. Maybe for another weekend...

File: server/controllers/stripeWebhook.js (hunk 4 above)
Comment (asood123) [inline server/controllers/stripeWebhook.js]: done

**Pull request merged**